src/main/java/seedu/address/logic/commands/EditEventCommand.java

```diff
@@ -0,0 +1,240 @@
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.event.DateTime;
+import seedu.address.model.event.Event;
+import seedu.address.model.person.Description;
+import seedu.address.model.person.Name;
+
+/**
+ * Edits the details of an existing event in the address book.
+ */
+public class EditEventCommand extends Command {
+
+    public static final String COMMAND_WORD = "editevent";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the person identified "
+            + "by the index number used in the displayed person list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NAME + "NEW_EVENT_NAME] "
+            + "[" + PREFIX_DATETIME + "NEW_DATE_TIME] "
+            + "[" + PREFIX_DESCRIPTION + "NEW_DESCRIPTION] "
+            + "[" + PREFIX_ADD_FRIENDNAME + "ADD_FRIEND_NAME1, ADD_FRIEND_NAME2…] "
+            + "[" + PREFIX_REMOVE_FRIENDNAME + "REMOVE_FRIEND_NAME1, REMOVE_FRIEND_NAME2…]...\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_NAME + "2nd Birthday "
+            + PREFIX_DATETIME + "16-08-2021 1600";
+
+    public static final String MESSAGE_EDIT_EVENT_SUCCESS = "Edited Event: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_EVENT = "This person already exists in the address book.";
+
+    public final Index index;
+    public final EditEventDescriptor editEventDescriptor;
+
+    /**
+     * @param index of the event in the event list to edit
+     * @param editEventDescriptor details to edit the event with
+     */
+    public EditEventCommand(Index index, EditEventDescriptor editEventDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editEventDescriptor);
+
+        this.index = index;
+        this.editEventDescriptor = new EditEventDescriptor(editEventDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Event> lastShownList = model.getEventsList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Event eventToEdit = lastShownList.get(index.getZeroBased());
+        Event editedEvent = createEditedEvent(eventToEdit, editEventDescriptor);
+
+        if (!eventToEdit.isSameEvent(editedEvent) && model.hasEvent(editedEvent)) {
```

Comment:
Just a quick check, what if the user types in the same event name as the event at the specified index for editevent command?